File: 100-Days-Series/Day 21/FibonacciUsingRecursion.java
public class FibonacciUsingRecursion {
    // Recursive method to calculate Fibonacci numbers
    public static int fibonacci(int n) {
        if (n <= 1) {
            return n; // Base cases: Fibonacci(0) = 0, Fibonacci(1) = 1
        }
        return fibonacci(n - 1) + fibonacci(n - 2); // Recursive call
    }

    public static void main(String[] args) {
        int n;
        java.util.Scanner s1=new java.util.Scanner(System.in);
        System.out.print("Enter a number  : ");
        n=s1.nextInt();
        // Number of terms in the Fibonacci series
        System.out.println("Fibonacci series up to " + n + " terms:");
        for (int i = 0; i < n; i++) {
            System.out.print(fibonacci(i) + " ");
        }
    }
}

File: 100-Days-Series/Day 21/FibonacciUsingRecursion.c
#include<stdio.h>
   // Recursive method to calculate Fibonacci numbers
int getFabonacci(int num){
    if(num<=1)
        return num;// Base cases: Fibonacci(0) = 0, Fibonacci(1) = 1
    else
        return getFabonacci(num-1)+getFabonacci(num-2);// Recursive call
}
void main(){
    int num;
    // Number of terms in the Fibonacci series
    printf("\nEnter series terms : ");
    scanf("%d",&num);
    printf("Fibonacci series up to %d terms:",num);
    for(int i=0;i<num;i++){
        printf("%d " ,getFabonacci(i));
    }
}